src/daft-core/src/array/ops/list.rs

```diff
@@ -42,6 +43,80 @@ fn join_arrow_list_of_utf8s(
         })
 }
 
+// Given an i64 array that may have either 1 or `self.len()` elements, create an iterator with
+// `self.len()` elements. If there was originally 1 element, we repeat this element `self.len()`
+// times, otherwise we simply take the original array.
+fn create_iter<'a>(arr: &'a Int64Array, len: usize) -> Box<dyn Iterator<Item = i64> + '_> {
+    match arr.len() {
+        1 => Box::new(repeat(arr.get(0).unwrap()).take(len)),
+        arr_len => {
+            assert_eq!(arr_len, len);
+            Box::new(arr.as_arrow().iter().map(|x| *x.unwrap()))
+        }
+    }
+}
+
+pub fn get_slices_helper(
+    parent_offsets: &arrow2::offset::OffsetsBuffer<i64>,
```

Comment:
You can actually make this an `impl Iterator<Item = i64>` (since you mostly use it that way anyways!) and then the fixed size list case would be quite clean: `std::iter::repeat(element_len).take(list_len)`

Reply:
Done, thanks for the suggestion!